### DISCUSSION TITLE
[Feature] Self-hosted text-to-speech support

### DISCUSSION THREAD

**Discussion by defaultsecurity** [open-webui/open-webui#2168] [General]
Support for self-hosted (local and remote) text-to-speech would be awesome and private.

Here are a few possibilities:

### XTTS or TTS by Coqui
- https://huggingface.co/coqui/XTTS-v2
- https://github.com/coqui-ai/TTS

### Mimic 3 by MyCroft
- https://github.com/MycroftAI/mimic3

### Piper by Rhasspy
- https://github.com/rhasspy/piper/

**Comment by justinh-rahb** [comment]
There's a lot of duplicates for this request.

**Comment by pazoff** [comment]
Samples for [Piper](https://github.com/rhasspy/piper), a fast and local text to speech system:
https://rhasspy.github.io/piper-samples/

**Comment by edurenye** [comment]
For TTS, STT and wake words we could integrate something like the [Wyoming Protocol](https://github.com/rhasspy/wyoming) allowing us to connect it to many TTS like Whisper or Vosk, STT like Piper and Wake word detection software like openWakeWord, porcupine1 or snowboy.

**Comment by defaultsecurity** [comment]
I have found a solution that produces acceptable voices:
https://github.com/matatonic/openedai-speech

It is based on XTTSv2 and provides OpenAI endpoints that Open WebUI handles perfectly.
